src/lib.rs

```diff
@@ -125,7 +119,15 @@ pub fn disasm<P>(path: P) -> Result<(), Error>
     let buf = unsafe { memmap::Mmap::map(&f)? };
 
     let obj = object::File::parse(&*buf).map_err(|e| DisasmError::Object { reason: e } )?;
-    disasm_sections(&obj, path)
+    if obj.has_debug_symbols() {
+        disasm_sections(&obj, &obj)
+    } else {
+        let debug_path = moria::locate_debug_symbols(&obj, path)?;
+        let debug_f = File::open(debug_path)?;
+        let debug_buf = unsafe { memmap::Mmap::map(&debug_f)? };
+        let debug_obj = object::File::parse(&*debug_buf).map_err(|e| DisasmError::Object { reason: e } )?;
```

Comment:
This probably wants to wind up as a helper function eventually.

Reply:
Yeah, it could be awkward due to the lifetimes though.